rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -212,4 +213,120 @@ Node::create_service(
   return serv;
 }
 
+const std::vector<rcl_interfaces::SetParametersResult>
+Node::set_parameters(
+  const std::vector<rcl_interfaces::Parameter> & parameters)
+{
+  std::vector<rcl_interfaces::SetParametersResult> results;
+  for (auto p : parameters) {
+    parameters_[p.name] = rclcpp::parameter::ParameterVariant::from_parameter(p);
+    rcl_interfaces::SetParametersResult result;
+    result.successful = true;
+    // TODO: handle parameter constraints
+    results.push_back(result);
+  }
+  return results;
+}
+
+const rcl_interfaces::SetParametersResult
+Node::set_parameters_atomically(
+  const std::vector<rcl_interfaces::Parameter> & parameters)
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+  std::map<std::string, rclcpp::parameter::ParameterVariant> tmp_map;
+  for (auto p : parameters) {
+    tmp_map[p.name] = rclcpp::parameter::ParameterVariant::from_parameter(p);
+  }
+  tmp_map.insert(parameters_.begin(), parameters_.end());
```

Comment:
Should this line go before the preceding for loop? It seems like old values will overwrite the new ones you just converted and stuck in the map.